storage/remote/queue_manager.go

```diff
@@ -884,11 +888,63 @@ outer:
 			t.metrics.droppedHistogramsTotal.WithLabelValues(reasonTooOld).Inc()
 			continue
 		}
-		if t.protoMsg == config.RemoteWriteProtoMsgV1 && h.FH != nil && h.FH.Schema == histogram.CustomBucketsSchema {
-			// We cannot send native histograms with custom buckets (NHCB) via remote write v1.
-			t.metrics.droppedHistogramsTotal.WithLabelValues(reasonNHCBNotSupported).Inc()
-			t.logger.Warn("Dropped float native histogram with custom buckets (NHCB) as remote write v1 does not support itB", "ref", h.Ref)
-			continue
+		if t.convertNHCBToClassic {
+			// we make labels here :D
+			t.seriesMtx.Lock()
+			lbls, ok := t.seriesLabels[h.Ref]
+			if !ok {
+				if _, ok := t.droppedSeries[h.Ref]; !ok {
+					t.logger.Info("Dropped histogram for series that was not explicitly dropped via relabelling", "ref", h.Ref)
+					t.metrics.droppedHistogramsTotal.WithLabelValues(reasonUnintentionalDroppedSeries).Inc()
+				} else {
+					t.metrics.droppedHistogramsTotal.WithLabelValues(reasonDroppedSeries).Inc()
+				}
+				t.seriesMtx.Unlock()
+				continue
+			}
+			meta := t.seriesMetadata[h.Ref]
+			t.seriesMtx.Unlock()
+			// we convert the NHCB to classic here.
+			// add labels for the buckets.
+			classicHistogram, err := histogram.ConvertNHCBToClassicHistogram(h.FH)
+			if err != nil {
+				t.logger.Error("Dropped histogram for series due to conversion error", "ref", h.Ref)
+				t.metrics.droppedHistogramsTotal.WithLabelValues(reasonUnintentionalDroppedSeries).Inc()
+			} else {
+				baseName := lbls.Get("__name__")
+				// we need add labels for the buckets.
+				// enqueue it
+				for _, bucket := range classicHistogram.Buckets {
```

Comment:
Instead of converting float to temp (is it really useful conversion? Would it be too different if we convert from float/int directly?) can we have a helper function that e.g iterate and spits series? 

Perhaps more useful interface would be  

```
func ConvertNHCBToClassicHistogram(nhcb *FloatHistogram, bucketSeries func(lset labels.Labels, v float64)) error
```

or we got fancy and do Go iterators (https://www.bwplotka.dev/2025/go-grpc-inprocess-iter/) or [Prometheus iterator](https://github.com/prometheus/prometheus/blob/d7e9a2ffb0f0ee0b6835cda6952d12ceee1371d0/tsdb/chunkenc/float_histogram.go#L142), but simple function calls would be fine for now.